lib/pbench/agent/tool_meister.py

```diff
@@ -248,162 +348,133 @@ def wait(self):
-                    "Transient tool %s start process failed with %d", self.name, sts
-                )
-        self.start_process = None
+        # First wait for the "stop" process to do it's job ...
+        self.stop_process = self._wait_for_process_w_kill(self.stop_process, "stop")
+        # ... then we wait for the start process to finish; in either case,
+        # we'll only wait a short time before killing them.
+        self.start_process = self._wait_for_process_w_kill(self.start_process, "start")
 
 
-class PcpTransTool(Tool):
-    """PcpTransTool - A more traditional alternative to the pcp persistent tool that
-                      allows one to run both the pmlogger and pmcd locally on each
-                      registered node. Additionally, this tool starts, stops, and
-                      sends data alongside other transient tools, rather than always
-                      running in the background.
+class PcpTransientTool(Tool):
+    """The transient tool alternative to the PCP persistent tool, which starts
+    and stops both a pmcd and pmlogger process, and sends data remotely, as
+    directed for transient tools.
     """
 
-    def __init__(self, name, tool_opts, logger=None, **kwargs):
-        super().__init__(name, tool_opts, logger=logger, **kwargs)
-        if self.tool_dir:
-            self.tool_dir = self.tool_dir / self.name.replace("-transient", "")
+    def __init__(self, name, tool_opts, **kwargs):
+        super().__init__(name, tool_opts, **kwargs)
+        self.tool_inst_dir = self.tool_dir / self.name.replace("-transient", "")
         if "/usr/libexec/pcp/bin" not in os.environ["PATH"]:
-            os.environ["PATH"] += os.pathsep + "/usr/libexec/pcp/bin"
-        pmcd_path = find_executable("pmcd")
-        if pmcd_path:
-            self.pmcd_args = [
-                pmcd_path,
-                "--foreground",
-                "--socket=./pmcd.socket",
-                "--port=55677",
-                f"--config={self.pbench_install_dir}/templates/pmcd.conf",
-            ]
-        else:
-            self.pmcd_args = None
-        pmlogger_path = find_executable("pmlogger")
-        if pmlogger_path:
-            self.pmlogger_args = [
-                pmlogger_path,
-                "--log=-",
-                "--report",
-                "-t",
-                "3s",
-                "-c",
-                f"{self.pbench_install_dir}/templates/pmlogger.conf",
-                "--host=localhost:55677",
-                f"{self.tool_dir}/%Y%m%d.%H.%M",
-            ]
-        else:
-            self.pmlogger_args = None
+            # FIXME - Shouldn't this be provided by the environment?
+            os.environ["PATH"] += ":/usr/libexec/pcp/bin"
+        self.pmcd_args = None
         self.pmcd_process = None
+        self.pmlogger_args = None
         self.pmlogger_process = None
-        if self.tool_dir:
-            try:
-                self.tool_dir.mkdir()
```

Comment:
Where/how does the tool directory get created now?

Reply:
This is a `PcpTransTool` so I believe the `tool_dir` creation should happen in `ToolMeister.start` call

Reply:
@npalaska, I'm not finding a `ToolMeister.start()` function.  The class has a `start_tools()` method -- did you mean that?  That does create a directory which it passes to the tool that it's starting.

@portante, so, the `mkdir()` call which was removed here was redundant (and presumably resulting in errors)?

FWIW, to my eye, the directory-creation code in `ToolMeister.init_tools()` and `ToolMeister.start_tools()` looks tragically similar (although `start_tools()` has better documentation)...should it be refactored into a helper function?  Also, I think the code recurs in `ToolMeister.sysinfo()`.  Having that level of duplication is really discomfiting.

Reply:
Regarding @webbnh's question, yes.  We just don't have an existing test for it.

Reply:
Can you comment on my suggestion to refactor the directory-creation code?  (I.e., tell me that it's already taken care of in #2697!  😆)

Reply:
Yes, I believe it is taken care of in PR #2697.